Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -26,11 +28,19 @@ internal class DependencyCollectorDiagnosticListener : IObserver<DiagnosticListe
         private readonly bool setComponentCorrelationHttpHeaders;
         private readonly IEnumerable<string> correlationDomainExclusionList;
         private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
-        private readonly ConcurrentDictionary<Guid, DependencyTelemetry> pendingTelemetry = new ConcurrentDictionary<Guid, DependencyTelemetry>();
+
+        private readonly ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>> pendingTelemetry = new ConcurrentDictionary<string, IOperationHolder<DependencyTelemetry>>();
 
         internal DependencyCollectorDiagnosticListener(TelemetryConfiguration configuration, bool setComponentCorrelationHttpHeaders = true, IEnumerable<string> correlationDomainExclusionList = null, ICorrelationIdLookupHelper correlationIdLookupHelper = null)
         {
+            configuration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+
             this.client = new TelemetryClient(configuration);
+            if (string.IsNullOrEmpty(this.client.InstrumentationKey))
+            {
+                this.client.InstrumentationKey = configuration.InstrumentationKey;
```

Comment:
no, please do not do it. Customer may override the `configuration.InstrumentationKey` for the `TelemetryConfiguration.Active` later than this module will be initialized